homework/unique_ptr/my_unique_ptr.hpp

```diff
@@ -1,40 +1,45 @@
 #pragma once
 
-template <typename T> class my_unique_ptr {
+template <typename T>
+class my_unique_ptr
+{
 
 public:
-  explicit my_unique_ptr(T *ptr) : ptr_(ptr){};
+    explicit my_unique_ptr(T* ptr) : ptr_(ptr){};
 
-  my_unique_ptr() : ptr_(nullptr) {}
+    my_unique_ptr() : ptr_(nullptr) {}
 
-  my_unique_ptr(my_unique_ptr<T> &&ptr_m) : ptr_{ptr_m.release()} {}
+    my_unique_ptr(my_unique_ptr<T>&& ptr_m) : ptr_{ptr_m.release()} {}
 
-  T &operator=(my_unique_ptr<T> &&ptr_m) {
-    this->ptr_ = ptr_.release();
-    return *(this->ptr_);
-  }
+    T& operator=(my_unique_ptr<T>&& ptr_m)
+    {
+        this->ptr_ = ptr_m.release();
+        return *(this->ptr_);
+    }
```

Comment:
Valgrind claims that there are memory leaks in this move operator
https://github.com/marcindus/memory-management/runs/1269365447?check_suite_focus=true

You should delete a memory of the previously own object before assigning new value
And it should not return `T&`, but `my_unique_ptr<T>&`

```suggestion
    my_unique_ptr<T>& operator=(my_unique_ptr<T>&& ptr_m) noexcept
    {
        delete ptr_;
        ptr_ = ptr_m.release();
        return *this;
    }
```